src/uint/boxed/shl.rs

```diff
@@ -1,108 +1,127 @@
 //! [`BoxedUint`] bitwise left shift operations.
 
-use crate::{BoxedUint, CtChoice, Limb, Word};
+use crate::{BoxedUint, Limb};
 use core::ops::{Shl, ShlAssign};
 use subtle::{Choice, ConstantTimeLess};
 
 impl BoxedUint {
     /// Computes `self << shift`.
-    /// Returns zero if `shift >= Self::BITS`.
-    pub fn shl(&self, shift: u32) -> Self {
+    /// Returns `None` if `shift >= Self::BITS`.
```

Comment:
Looks like some copypasta: `BoxedUint` doesn't have a `Self::BITS`, but instead a dynamic `self.bits_precision()` (and there are several comments of this nature in this PR)

Reply:
Fixed